### PR TITLE
fix: validate new trigger form inline

### LINKED ISSUE #2992: Create a trigger flow appears to have multiple validation checks
## Describe the bug

When creating a new LUIS trigger it appears that it validates the .LU file while being created, but this feels aggressive as I am just following the expected flow to add this component. Then after submitting, it runs another validation checking for invalid characters.

## Version

Latest on master

## Browser

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [x] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a dialog with LUIS recognizer.
1. Create a new trigger and add a name with invalid chars. This immediately warns me that I have no utterances found - this is brand new.
image
1. Add the utterances and the error goes away. All appears well, select Submit,
image
1. A new error pops up that the name fails.
image
1. Fix the name. Select Submit and you are done!

## Expected behavior

It appears there are a couple of validators:
* .LU file has populated trigger and utterances
* Trigger name is valid

I would expect my input to either be validated as I am entering it **or** after I submitted it, but not a hybrid approach. If there is any live validation (.LU), I would expect the Submit button to be disabled until that passes.

### PR DESCRIPTION
**Pull request by liweitian**
## Description
Every field will show live errors now.
## Task Item
Closes #2992 
## Screenshots
aaaa

### CODE REVIEW

**Comment by liweitian** [comment]
@a-b-r-o-w-n It seems I can use useForm hook here. Will apply that after it is merged.

**Comment by cwhitten** [comment]
@liweitian please address conflicts on this

**Comment by liweitian** [comment]
> @liweitian please address conflicts on this

done

File: Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx [hunk 1]
```diff
@@ -133,11 +180,21 @@ export const TriggerCreationModal: React.FC<TriggerCreationModalProps> = (props)
...
+  };
+
   const onClickSubmitButton = (e) => {
     e.preventDefault();
-    const errors = validateForm(selectedType, formData, isRegEx, regexIntents);
 
-    if (Object.keys(errors).length) {
+    //If still have some errors here, it is a bug.
+    const errors = validateForm(selectedType, formData, isRegEx, regexIntents);
+    if (shouldDisable(errors)) {
```
[the remark this answers is not in the document]
Comment (liweitian) [reply Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx:197]: if (shouldDisable(errors)) is correct.
Ideally, the submit button will be disabled if there are errors, which means we do not need to check errors here. Like the comment says, if there are some errors here, it is a bug. But I found that the schema and some rules changed frequently which may cause some bugs. So I want to make sure user won't create something unexpected here.

And some more UT are added. I added e2e test cases days about trigger days ago.

[the remark this answers is not in the document]
Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx:197]: I see now. My fault. I didn't see enough context around this and got confused. Thanks for the reply.

**Pull request merged**